Create forgot-password.js

mekhtear opened the pull request.

mekhtear committed:
Create forgot-password.js

MmyTohsaka reviewed (changes requested):
For the DB operations, should we stay consistent and use the operations in the data/users.js file? There is a function for finding users by Email and we can create one there for updating the password as well.

MmyTohsaka committed:
Merge branch 'main' into mekhtear-patch-5

MmyTohsaka committed:
Checkpoint

- Added the recovery route to the routes/index.js
- Adjusted views to reference the new /recovery route

MmyTohsaka committed:
Checkpoint

- Modified forgot-password route to use DB operation functions in the /data directory
- Added new function in the users data file to update a user
- Modified forgot-password handlebar file to route to /recovery/ route

MmyTohsaka reviewed (approved):
I think it looks good with the DB function changes.
